src/panels/config/lovelace/dashboards/dialog-lovelace-dashboard-detail.ts

```diff
@@ -0,0 +1,293 @@
+          : html``}
+        <mwc-button
+          slot="primaryAction"
+          @click="${this._updateDashboard}"
+          .disabled=${urlInvalid || this._submitting}
+        >
+          ${this._params.dashboard
+            ? this.hass!.localize(
+                "ui.panel.config.lovelace.dashboards.detail.update"
+              )
+            : this.hass!.localize(
+                "ui.panel.config.lovelace.dashboards.detail.create"
+              )}
+        </mwc-button>
+      </ha-dialog>
+    `;
+  }
+
+  private _urlChanged(ev: PolymerChangedEvent<string>) {
+    this._error = undefined;
+    this._urlPath = ev.detail.value;
+  }
+
+  private _sidebarIconChanged(ev: PolymerChangedEvent<string>) {
+    this._error = undefined;
+    this._sidebarIcon = ev.detail.value;
+  }
+
+  private _sidebarTitleChanged(ev: PolymerChangedEvent<string>) {
+    this._error = undefined;
+    this._sidebarTitle = ev.detail.value;
+  }
+
+  private _fillUrlPath() {
+    if (this._urlPath) {
+      return;
+    }
+    const parts = this._sidebarTitle.split(" ");
+
+    if (parts.length) {
+      this._urlPath = parts[0].toLowerCase();
+    }
+  }
+
+  private _showSidebarChanged(ev: Event) {
+    this._showSidebar = (ev.target as HaSwitch).checked;
+  }
+
+  private _requireAdminChanged(ev: Event) {
+    this._requireAdmin = (ev.target as HaSwitch).checked;
+  }
+
+  private async _updateDashboard() {
+    this._submitting = true;
+    try {
+      const values: Partial<LovelaceDashboardMutableParams> = {
+        require_admin: this._requireAdmin,
+        sidebar: this._showSidebar
+          ? { icon: this._sidebarIcon, title: this._sidebarTitle }
+          : null,
+      };
+      if (this._params!.dashboard) {
+        await this._params!.updateDashboard(values);
+      } else {
+        (values as LovelaceDashboardCreateParams).url_path = this._urlPath.trim();
+        (values as LovelaceDashboardCreateParams).mode = this._confMode;
+        await this._params!.createDashboard(
+          values as LovelaceDashboardCreateParams
+        );
+      }
+      this._params = undefined;
+    } catch (err) {
+      this._error = err ? err.message : "Unknown error";
+    } finally {
+      this._submitting = false;
+    }
+  }
+
+  private async _deleteDashboard() {
+    this._submitting = true;
+    try {
+      if (await this._params!.removeDashboard()) {
+        this._params = undefined;
+      }
+    } finally {
+      this._submitting = false;
+    }
+  }
+
+  private _close(): void {
+    this._params = undefined;
+  }
+
+  static get styles(): CSSResult[] {
+    return [
+      css`
+        ha-dialog {
+          --mdc-dialog-min-width: 400px;
+          --mdc-dialog-max-width: 600px;
+          --mdc-dialog-title-ink-color: var(--primary-text-color);
+          --justify-action-buttons: space-between;
+        }
+        /* make dialog fullscreen on small screens */
+        @media all and (max-width: 450px), all and (max-height: 500px) {
```

Comment:
Is this not standardized ? I thought that we had some haDialogStyle resource.

Reply:
Yeah, this is for mwc dialog, the other for paper. But we should standardize it.